src/main/java/com/box/sdk/BoxAPIRequest.java

```diff
@@ -74,6 +74,12 @@ public BoxAPIRequest(BoxAPIConnection api, URL url, String method) {
         this.method = method;
         this.headers = new ArrayList<RequestHeader>();
         if (api != null) {
+            Map<String, String> customHeaders = api.getHeaders();
+            if (customHeaders != null) {
+                for (String header : customHeaders.keySet()) {
+                    this.addHeader(header, customHeaders.get(header));
+                }
+            }
             this.headers.add(new RequestHeader("X-Box-UA", api.getBoxUAHeader()));
```

Comment:
Minor - this can probably go as a header in the api connection as well instead of lingering around as a special value in the api connection